pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -194,30 +191,52 @@ def test_description(self) -> str:
     def predict_description(self) -> str:
         return "Predicting"
 
-    def setup(self, trainer, pl_module, stage: Optional[str] = None):
```

Comment:
Why remove it from the `setup` hook?

Reply:
This is for spawn support, putting this call in setup meant the progress bar was being created before the processes are being spawned. The progress bar is not pickleable, hence has to be made after the spawn of processes (this also follows inline of what happens for TQDM)